maestro-studio/web/src/api.ts

```diff
@@ -0,0 +1,79 @@
+import { DeviceScreen, Repl, ReplCommand } from './models';
+import useSWR, { mutate } from 'swr';
+
+export type ReplResponse = {
+  repl?: Repl | undefined
+  error?: any
+}
+
+export const wait = async (durationMs: number) => {
+  return new Promise(resolve => setTimeout(resolve, durationMs))
+}
+
+const makeRequest = async <T>(method: string, path: string, body?: Object | undefined): Promise<T> => {
+  const response = await fetch(path, {
+    method,
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    },
+    body: body ? JSON.stringify(body) : undefined,
+  })
+  if (!response.ok) {
+    const body = await response.text()
+    throw new Error(`Request failed ${response.status}: ${body}`)
+  }
+  return await response.json()
+}
+
+const useRepl = (): ReplResponse => {
+  const {data: repl, error} = useSWR<Repl>('/api/repl/watch')
```

Comment:
I think we need to pass `?currentVersion=<replVersion>` here for the backend not to throw

Reply:
This is just the key actually. It's not actually hitting the endpoint under the hood since there is not fetcher supplied here. I'm essentially just using swr as a cache for long polling results